frontend/udp/parser_test.go

```diff
@@ -0,0 +1,144 @@
+package udp
+
+import (
+	"bytes"
+	"testing"
+)
+
+func TestParseQuery(t *testing.T) {
+	passkey := "abc123"
+	buf := &bytes.Buffer{}
+
+	//now add optional parameters
```

Comment:
All of these tests are very similar in structure. They could probably be turned into a [table-driven tests](https://github.com/golang/go/wiki/TableDrivenTests) where the table looks like
```go
struct {
  header []byte
  urlData string
}{
{ []byte{0x2, 0x10}, "/?passkey=abc123" },
{ []byte{0x2, 0x0}, "" },
...
}
```